core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,7 +285,7 @@ private[spark] object Utils extends Logging {
         } else {
           // Restrict file permissions via chmod if available.
           // For Windows this step is ignored.
-          if (!isWindows && !chmod700(dir)) {
+          if (!isWindows && (conf==null || !isRunningInYarnContainer(conf)) && !chmod700(dir)) {
```

Comment:
this is pretty unreasonable. What if we just pass in a flag like `changePermissions: Boolean = true` and set this to false if we're running in YARN?